### PR TITLE
Improve lookml sql derived tables detection, add cascading derived tables to lineage

### PR DESCRIPTION
**Pull request by remisalmon**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

This PR:
- updates the sql-metadata dependency version that add supports to more SQL dialects and improve the detection of sql derived tables.
- add cascading derived tables to the lineage of lookml views (see https://docs.looker.com/data-modeling/learning-lookml/derived-tables#cascading_derived_tables)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/lookml.py [hunk 1]
```diff
@@ -383,7 +372,12 @@ def _load_model(self, path: str) -> LookerModel:
     def _construct_datalineage_urn(self, sql_table_name: str, connection: str) -> str:
         platform = self._get_platform_based_on_connection(connection)
 
-        if "." in platform:
+        # Check if table name matches cascading derived tables pattern (same platform)
+        if re.fullmatch(r"\w+\.SQL_TABLE_NAME", sql_table_name):
+            platform_name = self.source_config.platform_name
+            sql_table_name = sql_table_name.lower().split(".")[0]
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/lookml.py:378]: Is it possible that the table name is like `schema.tablename.SQL_TABLE_NAME`?

Comment (remisalmon) [reply metadata-ingestion/src/datahub/ingestion/source/lookml.py:378]: According to the Looker doc the format is `${derived_table_or_view_name.SQL_TABLE_NAME}` where `derived_table_or_view_name` refer to another lookml view.

See: https://docs.looker.com/data-modeling/learning-lookml/derived-tables#cascading_derived_tables and https://docs.looker.com/reference/view-params/sql_table_name-for-view

In the special case where
```
view: desired_view_name {
  sql_table_name: database_name.schema_name.table_name ;;
}
```
then `derived_table_or_view_name` should be equal to `desired_view_name` and not `sql_table_name`.

Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/lookml.py:378]: Ah I think that makes sense - thanks for clarifying

**Review by hsheth2 — APPROVED** [review]
LGTM!

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**